src/controller/java/src/chip/jsontlv/TlvToJson.kt

```diff
@@ -55,6 +55,37 @@ fun TlvReader.toJsonString(): String {
   }
   return getStructJson().toString()
 }
+
+/**
+ * Encodes TLV into kotlin Object. If the TLV reader is positioned TLV Structure, Object will return
+ * to json format.
+ */
+fun TlvReader.toAny(tag: Tag = AnonymousTag): Any? {
+  val element = peekElement()
+  val value = element.value
+  value.toAny()?.let {
+    skipElement()
+    return it
+  }
+  return when (value) {
+    is ArrayValue -> {
+      buildList {
+        enterArray(tag)
+        while (!isEndOfContainer()) {
```

Comment:
please consider to use !tlvReader.isEndOfTlv() && !(tlvReader.isEndOfContainer() && tlvReader.isLastElement()

https://github.com/project-chip/connectedhomeip/pull/28617

Reply:
I added. But, isLastElement API is not merged yet. In this reason, I added TODO flag.